forest/src/cli/genesis_cmd.rs

```diff
@@ -45,6 +63,62 @@ impl GenesisCommands {
                     }
                 }
             }
+            Self::AddMiner {
+                genesis_path,
+                preseal_path,
+            } => {
+                if let Err(err) = add_miner(genesis_path.to_string(), preseal_path.to_string()) {
+                    println!("Cannot add miner(s), error: {}", err)
+                };
+            }
+        }
+    }
+}
+
+fn add_miner(genesis_path: String, preseal_path: String) -> Result<(), Box<dyn std::error::Error>> {
+    let mut genesis_str = String::new();
+    File::open(&genesis_path)?.read_to_string(&mut genesis_str)?;
+    let mut template: GenesisTemplate = serde_json::from_str(&genesis_str)?;
+
+    let mut preseal_str = String::new();
+    File::open(preseal_path)?.read_to_string(&mut preseal_str)?;
+    let miners: HashMap<String, Miner> = serde_json::from_str(&preseal_str)?;
+
+    for (miner_address_str, miner) in miners.into_iter() {
+        println!("Adding miner {} to genesis template", miner_address_str);
+
+        let id = ACCOUNT_START + template.miners.len() as u64;
+
+        let maddress = match Address::from_str(&miner_address_str) {
+            Ok(addr) => addr,
+            Err(_) => {
+                println!("Can not parse miner's address: {}", miner_address_str);
```

Comment:
```suggestion
            Err(e) => {
                println!("Can not parse miner's address: {}: {}", miner_address_str, e);
```
might as well show the error